api-client/src/main/java/com/xing/api/CallSpec.java

```diff
@@ -318,6 +329,15 @@ private Builder(Builder<RT, ET> builder) {
             return this;
         }
 
+        public Builder<RT, ET> timeouts(int connectTimeout, int readTimeout) {
+            if (connectTimeout < 0 || readTimeout < 0) {
```

Comment:
Please also add check for 0, because if you set it to 0 means no-timeout at all.